controllers/etcd_controller.go

```diff
@@ -1293,15 +1293,20 @@ func (r *EtcdReconciler) removeDependantStatefulset(ctx context.Context, logger
 	if err = r.List(ctx, statefulSets, client.InNamespace(etcd.Namespace), client.MatchingLabelsSelector{Selector: selector}); err != nil {
 		return err
 	}
+
+	var errs []error
 	for _, sts := range statefulSets.Items {
 		if canDeleteStatefulset(&sts, etcd) {
-			logger.Info("Deleting statefulset", "statefulset", kutil.Key(sts.GetNamespace(), sts.GetName()).String())
-			if err := r.Delete(ctx, &sts); err != nil {
-				return err
+			var key = kutil.Key(sts.GetNamespace(), sts.GetName()).String()
+			logger.Info("Deleting statefulset", "statefulset", key)
+			if err := r.Delete(ctx, &sts, client.PropagationPolicy(metav1.DeletePropagationBackground)); err != nil {
+				errs = append(errs, fmt.Errorf("Error deleting statefulset %q: %s", key, err))
+			} else {
+				errs = append(errs, fmt.Errorf("Waiting for statefulset %q to be deleted", key))
```

Comment:
I think it's safe to directly return here w/o error aggregation. There should usually only be one statefulset, right?

It's also a little bit unfortunate that after this error, we will re-queue very quickly and often due to the nature of the rate limiting queue, i.e. a lot of errors will appear while the pod is not gone.

Reply:
> I think it's safe to directly return here w/o error aggregation. There should usually only be one statefulset, right?

Sure. I can simplify that.

> It's also a little bit unfortunate that after this error, we will re-queue very quickly and often due to the nature of the rate limiting queue, i.e. a lot of errors will appear while the pod is not gone.

I thought about it. One way to solve this would be to return a typed error if and check in the caller and use `RequeueAfter` for the specific type. WDYT?

Reply:
> I thought about it. One way to solve this would be to return a typed error if and check in the caller and use RequeueAfter for the specific type. Wasn't sure if it was worth the complexity, WDYT?

I alternatively, I can make `removeDependantStatefulset` return an additional boolean flag to indicate if deletion went through and requeue to check if the statefulset is gone should be after some time. This change might be simpler. WDYT?

Reply:
@amshuman-kr I feel the second option of using an additional boolean return value seems simpler for this problem, since it doesn't involve introducing a new error type just for this one occurrence of the error.

Reply:
Done in https://github.com/gardener/etcd-druid/pull/222/commits/5f8e801bc2f1e5dd670cc1edf968225bb7327293